servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ConnectionClosedAfterResponseTest.java

```diff
@@ -0,0 +1,268 @@
+import static io.netty.channel.ChannelOption.AUTO_CLOSE;
+import static io.netty.channel.ChannelOption.AUTO_READ;
+import static io.servicetalk.http.api.HttpHeaderNames.CONNECTION;
+import static io.servicetalk.http.api.HttpHeaderNames.CONTENT_LENGTH;
+import static io.servicetalk.http.api.HttpHeaderNames.TRANSFER_ENCODING;
+import static io.servicetalk.http.api.HttpHeaderValues.CHUNKED;
+import static io.servicetalk.http.api.HttpHeaderValues.CLOSE;
+import static io.servicetalk.http.api.HttpResponseStatus.OK;
+import static io.servicetalk.http.api.Matchers.contentEqualTo;
+import static io.servicetalk.http.netty.HttpProtocolConfigs.h1;
+import static io.servicetalk.transport.netty.internal.AddressUtils.localAddress;
+import static io.servicetalk.transport.netty.internal.BuilderUtils.serverChannel;
+import static io.servicetalk.transport.netty.internal.EventLoopAwareNettyIoExecutors.toEventLoopAwareNettyIoExecutor;
+import static io.servicetalk.transport.netty.internal.GlobalExecutionContext.globalExecutionContext;
+import static java.lang.Integer.MAX_VALUE;
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+
+public class ConnectionClosedAfterResponseTest {
+
+    @Rule
+    public final Timeout timeout = new ServiceTalkTestTimeout();
+
+    private final ServerSocketChannel server;
+    private final BlockingHttpClient client;
+    private final AtomicReference<CharSequence> encodedResponse = new AtomicReference<>();
+    private final CountDownLatch connectionClosedLatch = new CountDownLatch(1);
+
+    public ConnectionClosedAfterResponseTest() {
+        EventLoopAwareNettyIoExecutor eventLoopAwareNettyIoExecutor =
+                toEventLoopAwareNettyIoExecutor(globalExecutionContext().ioExecutor());
+        EventLoop loop = eventLoopAwareNettyIoExecutor.eventLoopGroup().next();
+
+        ServerBootstrap bs = new ServerBootstrap();
+        bs.group(loop);
+        bs.channel(serverChannel(loop, InetSocketAddress.class));
+        bs.childHandler(new ChannelInitializer() {
+            @Override
+            protected void initChannel(final Channel ch) {
+                ch.pipeline().addLast(new HttpRequestDecoder());
+                ch.pipeline().addLast(new HttpObjectAggregator(MAX_VALUE));
+                ch.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                    @Override
+                    public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
+                        if (msg instanceof FullHttpRequest) {
+                            ctx.writeAndFlush(ByteBufUtil.writeAscii(ctx.alloc(), encodedResponse.get()))
+                                    .addListener(ChannelFutureListener.CLOSE);
+                        }
+                        ctx.fireChannelRead(msg);
```

Comment:
IMHO it would be more correct to not call `fireChannelRead(...)` but `ReferenceCountUtil.release(msg)` and so consume it.